mbtools/inject_ib_ids.py

```diff
@@ -0,0 +1,42 @@
+import argparse
+from pathlib import Path
+from mbtools.models import MoodleHtmlElement
+from lxml import etree
+
+CLASSES_TO_INJECT = [
+    "os-raise-ib-input",
+    "os-raise-ib-pset",
+    "os-raise-ib-pset-problem"]
+
+
+def inject_ib_ids(html_dir):
+    all_files = []
+    for path in Path(html_dir).rglob('*.html'):
+        all_files.append(path)
+
+    for file_path in all_files:
+        with open(file_path, 'r') as f:
+            parent_string = '<content></content>'
+            parent_element = etree.fromstring(parent_string)
+            parent_element.text = f.read()
+            content = MoodleHtmlElement(parent_element, str(file_path))
```

Comment:
I don't think we need (or should) use the `MoodleHtmlElement` class here since this is intended to be used with free-standing HTML vs HTML inside of Moodle XML. For consistency with other scripts that are intended to be run post-extraction, perhaps we should parse things with bs4 as [used in fix-html](https://github.com/openstax/raise-mbtools/blob/e8a75dffb94ce66f5185fadb901b10ff3e362c46/mbtools/fix_html.py#L1). That will also help avoid things like [this](https://github.com/openstax/raise-mbtools/blob/e8a75dffb94ce66f5185fadb901b10ff3e362c46/mbtools/models.py#L471-L472) we've hit when serializing HTML with `lxml`.

Reply:
@rnathuji it appears that bs4 rearranges attributes into alphabetical order - I think this is just how it stores data. Are we ok with that? For example in [pset ib blocks](https://github.com/openstax/k12-apps-raise/blob/main/docs/schemas.md#schema-definition-3) the attributes will start with `data-button-text` then `data-content-id` ... and so on. I don't see a huge problem, but we should make a note of it.

Reply:
Yeah, these types of transformations are fine since they change nothing about the meaning of the markup. That's another reason to capture "complete" test cases since they then serve as documentation of what occurs (e.g. if we forget we can just go to the test case before / after and quickly see "ah, it'll do a, b, c").